src/gt4py/next/program_processors/runners/dace/workflow/decoration.py

```diff
@@ -67,27 +101,19 @@ def decorated_program(
                         if on_gpu
                         else ndarray.__array_interface__["data"][0]
                     )
-                    # backfill dictionary of arguments with field shape and stride symbols
-                    kwargs.update(sdfg_callable.get_field_domain_symbols(arg_name, arg.domain))
-                    kwargs.update(sdfg_callable.get_field_stride_symbols(arg_desc, ndarray))
-                else:
-                    assert isinstance(arg_desc, dace.data.Scalar)
-                    assert isinstance(last_call_args[i], ctypes._SimpleCData)
-                    actype = arg_desc.dtype.as_ctypes()
-                    last_call_args[i] = actype(arg)
-
-            return inp.fast_call()
-
-        sdfg_args = sdfg_callable.get_sdfg_args(
-            sdfg,
-            offset_provider,
-            *flat_args,
-            filter_args=False,
-            on_gpu=on_gpu,
-        )
-
-        with dace.config.temporary_config():
-            dace.config.Config.set("compiler", "allow_view_arguments", value=True)
-            return inp(**sdfg_args)
+                    # When we find an array we update the `this_call_args` map
+                    # with the shape and stride symbols that are associated to it.
+                    # Note that `inp.sdfg_arglist` was constructed from an ordered
+                    # dictionary whose order reflects the DaCe calling convention:
+                    # first the array arguments, then the scalar arguments.
+                    # Thus, when we enter the branch above for scalar arguments
+                    # we know that all arrays have been processed and all their
+                    # associated symbols have been added to `this_call_args`.
```

Comment:
```suggestion
                    # Because of the order `sdfg_arglist` is constructed (see above) the
                    #  arrays are found first. We will now update the values that are
                    #  associated to their shape and strides.  
```

Maybe it is possible to make it smoother.